### DISCUSSION TITLE
Why slots have to be contiguous on Zephyr ?

### DISCUSSION THREAD

**Discussion by ghost** [mcu-tools/mcuboot#1985] [Q&A]
Hi team !

The [Zephyr documentation](https://github.com/mcu-tools/mcuboot/blob/main/docs/readme-zephyr.md) states:

> Currently, the two image slots must be contiguous.

I would like to understand the reason behind this limitation.

## Context

I'm working on a Zephyr project on a STM32H573 with 2MB of flash organized in two banks of 1MB each. 

I would like to have my two slots to reside each on one flash bank, so I can benefit from the "read-while-write" mechanism when downloading my new FW.

![STM32H573i_dk_flash_partition](https://github.com/mcu-tools/mcuboot/assets/150900702/1f37dc37-7b68-4e5a-84bb-8466af4bb511)

**Comment by nordicjm** [comment]
It means the data in each slot must be contiguous, no that one slot must start right after the previous has ended, your setup there is ok
